src/nupic/algorithms/AnomalyLikelihood.hpp

```diff
@@ -0,0 +1,261 @@
+  :type verbosity: UInt
+
+  :returns: a vector of likelihoods, one for each aggregated point
+  **/
+    vector<Real>  updateAnomalyLikelihoods(vector<Real> anomalyScores, UInt verbosity=0);
+ /**
+  Given the normal distribution specified by the mean and standard deviation
+  in distributionParams (the distribution is an instance member of the class), 
+  return the probability of getting samples further from the mean. 
+  For values above the mean, this is the probability of getting
+  samples > x and for values below the mean, the probability of getting
+  samples < x. This is the Q-function: the tail probability of the normal distribution.
+  **/
+    Real32 tailProbability(Real32 x) const;
+  /**
+  :param sampleData:
+  :type sampleData: vector array //TODO of what? likelihoods? rawScores? ....?
+  :param performLowerBoundCheck:
+  :type performLowerBoundCheck: bool
+  :returns: A DistributionParams (struct) containing the parameters of a normal distribution based on
+      the ``sampleData``.
+  **/
+    DistributionParams estimateNormal(vector<Real> sampleData, bool performLowerBoundCheck=true);
+  /** returns parameters of a Null distribution **/
+    DistributionParams nullDistribution() const {
+    return DistributionParams("normal", 0.5, 1e6, 1e3);
+}
+
+    //private static methods
+static Real  computeLogLikelihood(Real likelihood)  {
+  /**
+    Compute a log scale representation of the likelihood value. Since the
+    likelihood computations return low probabilities that often go into four 9's
+    or five 9's, a log value is more useful for visualization, thresholding,
+    etc.
+   **/
+    // The log formula is:
+    //     Math.log(1.0000000001 - likelihood) / Math.log(1.0 - 0.9999999999)
+    return log(1.0000000001f - likelihood) / -23.02585084720009f;
+}
+
+static UInt calcSkipRecords_(UInt numIngested, UInt windowSize, UInt learningPeriod)  {
+    /** Return the value of skipRecords for passing to estimateAnomalyLikelihoods
+
+    If `windowSize` is very large (bigger than the amount of data) then this
+    could just return `learningPeriod`. But when some values have fallen out of
+    the historical sliding window of anomaly records, then we have to take those
+    into account as well so we return the `learningPeriod` minus the number
+    shifted out.
+
+    @param numIngested - (int) number of data points that have been added to the
+      sliding window of historical data points.
+    @param windowSize - (int) size of sliding window of historical data points.
+    @param learningPeriod - (int) the number of iterations required for the
+      algorithm to learn the basic patterns in the dataset and for the anomaly
+      score to 'settle down'.
+    **/
+    int diff = numIngested - (int)windowSize;
+    UInt numShiftedOut = max(0, diff);
+    return min(numIngested, max((UInt)0, learningPeriod - numShiftedOut));
+}
+
+
+static std::vector<Real> circularBufferToVector(boost::circular_buffer<Real> cb) {
```

Comment:
This shouldn't be in the header. (I know this will disappear when the `MovingWindow` is added, but I'm saying this again for completeness)